File: src/main/java/itmo/is/project/dto/module/storage/StorageModuleDto.java
package itmo.is.project.dto.module.storage;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

import java.util.List;

public record StorageModuleDto(

        @Schema(example = "1")
        @JsonProperty(value = "id", required = true)
        Integer id,

        @JsonProperty(value = "blueprint", required = true)
        StorageModuleBlueprintDto blueprint,

        @JsonProperty(value = "storedResources", required = true)
        List<StoredResourceDto> storedResources
) {
}

File: src/main/kotlin/itmo/isproject/storage/dto/StorageModuleDto.kt
package itmo.isproject.storage.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

data class StorageModuleDto(

    @Schema(example = "1")
    @JsonProperty(value = "id", required = true)
    val id: Int?,

    @JsonProperty(value = "blueprint", required = true)
    val blueprint: StorageModuleBlueprintDto,

    @JsonProperty(value = "storedResources", required = true)
    val storedResources: List<StoredResourceDto>
)
